third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -1009,18 +1009,33 @@ class ConvertSCFIfOp : public PointerCanonicalizationPattern<scf::IfOp> {
       rewriter.inlineBlockBefore(ifOp.elseBlock(), newIfOp.elseBlock(),
                                  newIfOp.elseBlock()->begin());
 
-    rewriter.replaceOpWithMultiple(ifOp, {newIfOp.getResults()});
-
-    for (int64_t idx :
-         llvm::cast<DenseI64ArrayAttr>(newIfOp.thenYield()->getDiscardableAttr(
-                                           kSCFIfOpYieldFatPtrOffsets))
-             .asArrayRef()) {
+    ArrayRef<long> yieldPtrOffsets =
+        llvm::cast<DenseI64ArrayAttr>(
+            newIfOp.thenYield()->getDiscardableAttr(kSCFIfOpYieldFatPtrOffsets))
+            .asArrayRef();
+    for (int64_t idx : yieldPtrOffsets) {
       Value thenFatPtrBase = newIfOp.thenYield().getOperand(idx);
       Value thenFatPtrOffset = newIfOp.thenYield().getOperand(idx + 1);
       fatPtrs[{newIfOp.getResult(idx), newIfOp.getResult(idx + 1)}] =
           fatPtrs.at({thenFatPtrBase, thenFatPtrOffset});
     }
 
+    ResultRange results = newIfOp.getResults();
+    SmallVector<ValueRange> replacements;
+    SetVector<int64_t> ptrIndices(yieldPtrOffsets.begin(),
+                                  yieldPtrOffsets.end());
+    int64_t idx = 0;
+    for (; idx < newIfOp.getNumResults();) {
+      if (ptrIndices.contains(idx)) {
+        replacements.push_back(results.slice(idx, 2));
```

Comment:
FYI don't index into `ResultRange`/`ValueRange` to create new `ResultRange`/`ValueRange` 🤦